lang/types/value.go

```diff
@@ -203,6 +203,182 @@ func ValueOf(v reflect.Value) (Value, error) {
 	}
 }
 
+// Into mutates the given reflect.Value with the data represented by the Value
+func Into(v Value, rv reflect.Value) error {
+	typ := rv.Type()
+	kind := typ.Kind()
+	for kind == reflect.Ptr {
+		typ = typ.Elem() // un-nest one pointer
+		kind = typ.Kind()
+
+		// if pointer was nil, instantiate the destination type and point
+		// at it to prevent nil pointer dereference when setting values
+		if rv.IsNil() {
+			rv.Set(reflect.New(typ))
+		}
+		rv = rv.Elem() // un-nest rv from pointer
+	}
+
+	// capture rv and v in a closure that is static for the scope of this Into() call
+	// mustInto ensures rv is in a list of compatible types before attempting to reflect it
+	mustInto := func(kinds ...reflect.Kind) {
+		// sigh. Go can be so elegant, and then it makes you do this
+		inlist := func(i reflect.Kind, l []reflect.Kind) bool {
+			for _, n := range l {
+				if i == n {
+					return true
+				}
+			}
+			return false
+		}
+		if !inlist(rv.Kind(), kinds) {
+			panic(fmt.Sprintf("cannot Into() a %s into reflect %s", v.Type(), kinds))
```

Comment:
The logic here is to show a clear divide between user errors, like the integer overflow errors, and type matching errors such as trying to `Into()` an int into a string container. I'm emulating what the reflect library would do if you tried to perform the equivalent action.
If the `Into()` function ever panics then it's certainly a programming error. Any type mismatches should be caught in the unification stage.

Reply:
> If the Into() function ever panics then it's certainly a programming error. Any type mismatches should be caught in the unification stage.

I completely agree! And it would be a bug. But for database-safety (and other reasons) when we hit this bug, an `error` will cause a safe shutdown, where as a panic would kill everything even if it's mid-write.

So for this reason we error.

It's better to panic in a situation before a bug that would otherwise *cause* a database corruption or some other bad thing. But then only if you can't bubble errors properly.

Reply:
Okay I'll replace all `panic()` with `return fmt.Errorf(..`

Reply:
Any calls to reflect will still panic though, if the errors make it that far, which for at least half of the types it will do. Should I try to catch the other errors too?

Reply:
It's true that the reflect lib panics in a number of cases. It's always good to check the panic condition if you know what it is when you can. Eg: if something will panic on calling a function with a nil arg, check if the arg is nil first. =D

Reply:
For the simple types: `SetInt`, `SetBool` etc I didn't add any type checks at all because I was relying on reflect causing the panics instead. Should I add some `mustInto()` checks for those too?

Reply:
Can't hurt AFAICT, right?

Reply:
A tiny amount of performance due to extra if statements, but should be fine